cmd/stack_ls.go

```diff
@@ -26,21 +27,47 @@ import (
 	"github.com/pulumi/pulumi/pkg/backend/display"
 	"github.com/pulumi/pulumi/pkg/backend/httpstate"
 	"github.com/pulumi/pulumi/pkg/backend/state"
-	"github.com/pulumi/pulumi/pkg/tokens"
 	"github.com/pulumi/pulumi/pkg/util/cmdutil"
 	"github.com/pulumi/pulumi/pkg/workspace"
 )
 
 func newStackLsCmd() *cobra.Command {
-	var allStacks bool
 	var jsonOut bool
+
+	var allStacks bool
+
+	var orgFilter string
+	var projFilter string
+	var tagFilter string
+
 	cmd := &cobra.Command{
 		Use:   "ls",
-		Short: "List all known stacks",
-		Args:  cmdutil.NoArgs,
+		Short: "List stacks",
+		Long: "List stacks\n" +
+			"\n" +
+			"This command lists stacks. By default only stacks with the same project name as the\n" +
+			"current workspace will be returned. By passing --all, all stacks you have access to\n" +
+			"will be listed.\n" +
+			"\n" +
+			"Results may be further filtered by passing additional flags. Tag filters may include\n" +
+			"the tag name as well as the tag value, separated by an equals sign. For example 'environment=production'\n" +
+			"or just 'devstack-owner'.",
+		Args: cmdutil.NoArgs,
 		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
-			var packageFilter *tokens.PackageName
-			if !allStacks {
+			// Build up the stack filters. If left empty, will be a no-op.
+			tagName, tagValue, err := parseTagFilter(tagFilter)
```

Comment:
Doesn't this always return an error when `tagFilter` is empty (which is the default?).  If so, should we only do this when tagFilter is non empty (or change parseTagFilter to return `"", "", nil` in that case)?